src/libraries/System.IO.Compression/src/System/IO/Compression/ZLibStream.cs

```diff
@@ -0,0 +1,293 @@
+        /// <param name="mode">One of the enumeration values that indicates whether to compress or decompress the stream.</param>
+        /// <param name="leaveOpen"><c>true</c> to leave the stream object open after disposing the <see cref="ZLibStream"/> object; otherwise, <c>false</c>.</param>
+        public ZLibStream(Stream stream, CompressionMode mode, bool leaveOpen)
+        {
+            _deflateStream = new DeflateStream(stream, mode, leaveOpen, ZLibNative.ZLib_DefaultWindowBits);
+        }
+
+        /// <summary>Initializes a new instance of the <see cref="ZLibStream"/> class by using the specified stream and compression level.</summary>
+        /// <param name="stream">The stream to which compressed data is written.</param>
+        /// <param name="compressionLevel">One of the enumeration values that indicates whether to emphasize speed or compression efficiency when compressing the stream.</param>
+        public ZLibStream(Stream stream, CompressionLevel compressionLevel) : this(stream, compressionLevel, leaveOpen: false)
+        {
+        }
+
+        /// <summary>Initializes a new instance of the <see cref="ZLibStream"/> class by using the specified stream, compression level, and whether to leave the <paramref name="stream"/> open.</summary>
+        /// <param name="stream">The stream to which compressed data is written.</param>
+        /// <param name="compressionLevel">One of the enumeration values that indicates whether to emphasize speed or compression efficiency when compressing the stream.</param>
+        /// <param name="leaveOpen"><c>true</c> to leave the stream object open after disposing the <see cref="ZLibStream"/> object; otherwise, <c>false</c>.</param>
+        public ZLibStream(Stream stream, CompressionLevel compressionLevel, bool leaveOpen)
+        {
+            _deflateStream = new DeflateStream(stream, compressionLevel, leaveOpen, ZLibNative.ZLib_DefaultWindowBits);
+        }
+
+        /// <summary>Gets a value indicating whether the stream supports reading.</summary>
+        public override bool CanRead => _deflateStream?.CanRead ?? false;
+
+        /// <summary>Gets a value indicating whether the stream supports writing.</summary>
+        public override bool CanWrite => _deflateStream?.CanWrite ?? false;
+
+        /// <summary>Gets a value indicating whether the stream supports seeking.</summary>
+        public override bool CanSeek => _deflateStream?.CanSeek ?? false;
+
+        /// <summary>This property is not supported and always throws a <see cref="NotSupportedException"/>.</summary>
+        public override long Length => throw new NotSupportedException();
+
+        /// <summary>This property is not supported and always throws a <see cref="NotSupportedException"/>.</summary>
+        public override long Position
+        {
+            get => throw new NotSupportedException();
+            set => throw new NotSupportedException();
+        }
```

Comment:
I can see that these are not supported because `DeflateStream` also does not support them. I was going to ask why they were throwing, but I assume we would've also tried to call the underlying `_deflateStream` properties if they didn't throw.
https://docs.microsoft.com/en-us/dotnet/api/system.io.compression.deflatestream.length?view=netcore-3.1
https://docs.microsoft.com/en-us/dotnet/api/system.io.compression.deflatestream.position?view=netcore-3.1